### DISCUSSION TITLE
Jerky Motion When Backlash Compensation Activated

### DISCUSSION THREAD

**Discussion by hanifhan** [grblHAL/core#584] [Q&A]
im trying use a backlash compensation fiture , and i found out that there is kind of jerky motion when im activated it , i assume that it happened because of acceleration or feedrate when the stepper move on compensation backlash is too slow or is it designed just like that ? here is my video of test

anyway , im compare the time of machining time 
- without backlash compensation : 10 minutes 30 second
- with backlash compensation X & Y : 13 minutes and 10 second

https://github.com/user-attachments/assets/bc728a72-1ff3-4c25-af2b-f20f9e57e26f

**Comment by zoidfarb-prints** [comment]
I'm also seeing this, did you ever figure out a solution?

**Reply by terjeio** [reply]
IMO the only practical solution is to get rid of the backlash mechanically.
Taking out backlash has to be done by reversing the motion until the nut/gears engages again. This introduces a motion vector which is 180 degrees from the original - which affects speed and acceleration calculations negatively. It might be possible to spread out the backlash compensation move over several arc segments, perhaps someone could try to implement that?
